applications/nrf_desktop/doc/ble_discovery.rst

```diff
@@ -1,6 +1,75 @@
+|                              |                                  |                   |                                  +--------------------------------+
+|                              |                                  |                   |                                  | ``ble_discovery``              |
++------------------------------+----------------------------------+-------------------+----------------------------------+--------------------------------+
+
+Configuration
+*************
+
+Complete the following steps to configure the module:
+
+1. Complete the basic Bluetooth configuration, as described in the Bluetooth guide.
+#. Set the :option:`CONFIG_BT_GATT_CLIENT` Kconfig option to enable support for the GATT Client role.
+#. Set the :option:`CONFIG_BT_GATT_DM` Kconfig option to enable the :ref:`gatt_dm_readme`.
+   The :ref:`gatt_dm_readme` is used by the ``ble_discovery`` application module.
+#. Define the module configuration in the ``ble_discovery_def.h`` file, located in the board-specific directory in the application configuration folder.
+   You must define the following parameters of every nRF Desktop peripheral that connects with the given nRF Desktop central:
+
+      * Vendor ID (VID) - common for all the peripherals.
+      * Product ID (PID) and peer type (:cpp:enum:`PEER_TYPE_MOUSE` or :cpp:enum:`PEER_TYPE_KEYBOARD`) - defined separately for every peripheral.
+
+For an example of the module configuration, see :file:`configuration/nrf52840_pca10059/ble_discovery_def.h`.
+
+This module configuration example uses 0x1915 as Nordic Semiconductor vendor ID, ensure to change this to correspond to the vendor ID of your company.
+The assinged product IDs should be unique among devices using given vendor ID.
+Otherwise host will be unable to distinguish the devices.
+
+#. Set the ``CONFIG_DESKTOP_BLE_DISCOVERY_ENABLE`` Kconfig option to enable the ``ble_discovery`` application module.
+
+Implementation details
+**********************
+
+Peripheral discovery
+   The module starts the peripheral device discovery when it receives ``ble_peer_event`` with :cpp:member:`state` set to :cpp:enum:`PEER_STATE_SECURED`.
+
+   The peripheral discovery consists of the following steps:
+
+   * Reading device description
+      The central checks if the connected peripheral supports the LLPM (Low Latency Packet Mode).
+      The device description is a GATT Service that is specific for nRF Desktop peripherals and is implemented by :ref:`nrf_desktop_dev_descr`.
+   * Reading DIS (Device Information Service)
+      The central reads VID and PID of the connected peripheral.
+   * HIDS (Human Interface Device Service) discovery
+      The central discovers HIDS and forwards the information to other application modules using ``ble_discovery_complete`` event.
+      The :ref:`nrf_desktop_hid_forward` uses the event to register a new subsriber.
```

Comment:
This ref is reported as broken in the local build...

Reply:
I rebased the PR.